clean_helpers/deduplication.py

```diff
@@ -81,14 +82,36 @@ def dedup_document(ds: Dataset, num_proc: int, batch_size: int) -> Dataset:
     #     batch_size=batch_size,
     # ).remove_columns("hash")
 
-url_regex = re.compile(r"^(.[^?]*)")
+def simplify_url(url: str) -> str:
+    # We only keep the query parameter of type id in the url
+
+    url_parse = urllib.parse.urlparse(url)
+    query = url_parse.query
+
+    url_dict = dict(urllib.parse.parse_qsl(query))
+
+    # We keep only the query parameters of type id
+    if "id" in url_dict:
+        url_dict = {"id": url_dict["id"]}
+    elif "news-id" in url_dict:
+        # Specific treatment to remap "news-id" as an "id" query param
+        url_dict = {"id": url_dict["news-id"]}
+    else:
+        url_dict = {}
```

Comment:
We can actually expose this mapping as a parameter to the builder.
```
# store the name of the parameter in the url as key and it's value is it's normalised version.
parameters_map = {
   "id": "id",
   "news-id": "id"
}

def build_dedup_document_on_url(parameters_map: Dict[str, str]):
   blablabla
```
The idea is we can build dataset specific values, instead of having a one size fit all type of mechanism.